sensors/sensors.c

```diff
@@ -0,0 +1,600 @@
+	rbnode_t *node;
+	const sensor_info_t *info;
+
+	if (devsz <= 0) {
+		fprintf(stderr, "sensors: wrong device number\n");
+		return -EINVAL;
+	}
+
+	/* Initialize device table */
+	sensors_common.devEvents = malloc(devsz * sizeof(uint8_t *));
+	if (sensors_common.devEvents == NULL) {
+		fprintf(stderr, "sensors: cannot allocate memory\n");
+		return -ENOMEM;
+	}
+
+	/* Each device can publish data of all sensor types. The following table contains information about device's events positions
+	   in sensors_common.events tables */
+	for (i = 0; i < devsz; ++i) {
+		sensors_common.devEvents[i] = calloc(1, NB_SENSOR_TYPES);
+		if (sensors_common.devEvents == NULL) {
+			fprintf(stderr, "sensors: cannot allocate memory\n");
+			return -ENOMEM;
+		}
+	}
+
+	/* Assign position of events for each device and calculate number of events of all types */
+	for (node = lib_rbMinimum(sensors_common.infos.root), i = 0; node != NULL; node = lib_rbNext(node), ++i) {
+		info = lib_treeof(sensor_info_t, node, node);
+
+		for (j = 0; j < NB_SENSOR_TYPES; ++j) {
+			id = 1 << j;
+			if ((info->types & id) != 0) {
+				sensors_common.devEvents[i][j] = sensors_common.evtNb[j]++;
+			}
+		}
+	}
+
+	/* Allocate memory for events tables */
+	for (i = 0; i < NB_SENSOR_TYPES; ++i) {
+		if (sensors_common.evtNb[i] != 0) {
+			sensors_common.events[i] = malloc(sizeof(sensor_event_t) * sensors_common.evtNb[i]);
+			if (sensors_common.events[i] == NULL) {
+				fprintf(stderr, "sensors: cannot allocate memory for events\n");
+				continue;
+			}
+
+			res = mutexCreate(&sensors_common.locks[i]);
+			if (res < 0) {
+				free(sensors_common.events[i]);
+				fprintf(stderr, "sensors: cannot create lock\n");
+			}
+		}
+	}
+
+	return EOK;
+}
+
+
+static int sensors_drvInit(const char *name, const char *args)
+{
+	int res;
+	sensor_info_t *info;
+	const sensor_drv_t *drv;
+
+	drv = sensors_getDrv(name);
+	if (drv == NULL) {
+		fprintf(stderr, "sensors: cannot find driver for %s\n", name);
+		return -EINVAL;
+	}
+
+	info = malloc(sizeof(sensor_info_t));
+	if (info == NULL) {
+		fprintf(stderr, "sensors: cannot allocate memory\n");
+		return -ENOMEM;
+	}
+
+	res = drv->alloc(info, args);
+	if (res < 0) {
+		free(info);
+		fprintf(stderr, "sensors: cannot allocate sensor %s, err: %d\n", name, res);
+		return res;
+	}
+
+	res = idtree_alloc(&sensors_common.infos, &info->node);
+	if (res < 0) {
+		free(info);
+		fprintf(stderr, "sensors: cannot add sensor %s, to tree - err: %d\n", name, res);
+		return res;
+	}
+
+	info->id = res;
+	info->drv = drv->name;
+
+	return res;
+}
+
+
+static int sensors_parseArgs(int argc, char **argv)
+{
+	int c, devsz = 0;
+	char *args;
+	const char *name;
+
+	while ((c = getopt(argc, argv, "s:h")) != -1) {
+		switch (c) {
+			case 's': /* <sensor:args> */
+				name = optarg;
+				args = strchr(optarg, ':');
+				if (args == NULL) {
+					fprintf(stderr, "sensors: missing a sensors arguments: %s\n", optarg);
+					return devsz;
+				}
+
+				*args++ = '\0';
```

Comment:
Imho better make operation order explicit 
```suggestion
				*(args++) = '\0';
```